### ISSUE TITLE
Error on launch after upgrade to 2.0.6 (Immediate attempt to place order)

### ISSUE THREAD

**Issue by rcarmo** [whittlem/pycryptobot#161]
**Describe the bug**

I restarted my bots after upgrading to 2.0.6 and got an immediate error that appears to be tied to the bot trying to place an immediate order without checking for funds (maybe wrong calculations?). There are no manual orders, but there are also no funds for the bot to try to place the order, so I'm wondering why it's trying in the first place.

**To Reproduce**
1. Run bot

**Expected behavior**
Should have run and checked for funds before placing the order.

**Desktop (please complete the following information):**
 - OS: Linux

**Additional context**

Traceback:
```
Traceback (most recent call last):
  File "../../pycryptobot.py", line 61, in <module>
    raise Exception('Insufficient available funds to place buy order: ' + str(account.getBalance(app.getQuoteCurrency())) + ' < 50 ' + app.getQuoteCurrency() + "\nNote: A manual limit order places a hold on available funds.")
Exception: Insufficient available funds to place buy order: 0.01 < 50 EUR
Note: A manual limit order places a hold on available funds.
```

**Comment by whittlem** [comment]
It sounds like the bot is having a problem detecting what the last action was... either a buy or sell. If your next action is a buy then start the bot with --lastaction SELL and if the next action is a sell then start with --lastaction BUY. Did you make any changes to your portfolio? Maybe a manual buy or sell or changing the balance?

**Comment by rcarmo** [comment]
No, it’s al automated. No manual changes to the portfolio. But there are previous filled orders, so I’m wondering why it can’t figure that out.

In this particular case, the last order was a BUY, which makes it all the more strange - I don’t know why it would go for another BUY, as the previous BUY was at a higher price than current market conditions on startup.

**Comment by rcarmo** [comment]
Just tried --lastaction BUY and still have the same error. Upgraded from git tip, same issue.

**Comment by whittlem** [comment]
I will look, thanks.

**Comment by whittlem** [comment]
I think a PR broke this in the morning. Can you confirm yours is working now?

**Comment by whittlem** [comment]
Can you confirm this issue is resolved after the last updates?

**Comment by p9qgtjaliucdtygqapv79h5nm1faov8** [comment]
>  but there are also **no funds** for the bot to try to place the order, so I'm wondering why it's trying in the first place.

No funds as in no quote currency and some crypto or no funds as in nothing in the account?

If there are no funds in the account than this is a repeat of issue #36.

Around line 60 the bot is not trying to make an order. This is where the bot checks if there are enough funds to run. The only time I have experienced this is when I tried to run the bot with insufficient funds (ex. less than $50).

**Comment by whittlem** [comment]
I've just merged a PR from someone who said they have improved this. Can you try it now and see if it works for you?

**Comment by rcarmo** [comment]
Can't reproduce this anymore, as I changed my setup and portfolio.

**Comment by iThom** [comment]
Reporter can't reproduce the issue and many fix were done since v2.0.6 (we are currently in v2.13.5).
Let's reopen this issue if problem appears again.
